pm/sender.go

```diff
@@ -79,17 +79,20 @@ func (s *sender) EV(sessionID string) (*big.Rat, error) {
 	return ticketEV(session.ticketParams.FaceValue, session.ticketParams.WinProb), nil
 }
 
-func (s *sender) validateSender() error {
-	info, err := s.senderManager.GetSenderInfo(s.signer.Account().Address)
-	if err != nil {
-		return ErrSenderValidation{fmt.Errorf("unable to validate sender: could not get sender info: %v", err)}
-	}
-
+func (s *sender) validateSender(info *SenderInfo) error {
 	maxWithdrawRound := new(big.Int).Add(s.timeManager.LastInitializedRound(), big.NewInt(1))
 	if info.WithdrawRound.Int64() != 0 && info.WithdrawRound.Cmp(maxWithdrawRound) != 1 {
 		return ErrSenderValidation{fmt.Errorf("unable to validate sender: deposit and reserve is set to unlock soon")}
 	}
 
+	if info.Reserve.FundsRemaining.Int64() == 0 {
+		return ErrSenderValidation{errors.New("no sender reserve")}
+	}
+
+	if info.Deposit.Int64() == 0 {
+		return errors.New("no sender deposit")
```

Comment:
Why not wrap this as a `ErrSenderValidation` error since no transcoding will happen until B has a non-zero deposit anyway?

Reply:
fixed in facbd57